src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Action;
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransactions(List<String> dateList) {
+        List<FruitTransaction> result = new ArrayList<>();
+        for (String string : dateList) {
+            String[] stringArray = string.split(",");
+            Action action = Action.fromCode(stringArray[0]);
+            String fruit = stringArray[1];
+            int quantity = Integer.parseInt(stringArray[2]);
+
+            result.add(new FruitTransaction(action, fruit, quantity));
+        }
+        return result;
+    }
```

Comment:
It's a good practice to validate the input data before processing it. You should check if `dateList` is not null and not empty to prevent `NullPointerException` or unexpected behavior.